ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSpec.java

```diff
@@ -279,6 +280,17 @@ public void setIsMetadataOnly(boolean isMetadataOnly){
     this.isMetadataOnly = isMetadataOnly;
   }
 
+  /**
+   * @return true if this statement refers to metadata-only operation.
+   */
+  public boolean isMetadataOnlyForExternalTables() {
```

Comment:
i think the only change for this patch should be ..if its external table metadata only ..then just dont dump external table data